### PR TITLE
[BPK-509] Add dismissable configuration

### PR DESCRIPTION
**Pull request by georgegillams**

### CODE REVIEW

File: packages/bpk-component-banner-alert/stories.js [hunk 1]
```diff
@@ -39,8 +40,14 @@ storiesOf('bpk-component-banner-alert', module)
   .add('Success (long message)', () => (
     <BpkBannerAlert message={longMessage} type={ALERT_TYPES.SUCCESS} />
   ))
+  .add('Success (dismissable)', () => (
+    <BpkBannerAlert dismissable message={message} type={ALERT_TYPES.SUCCESS} />
```
Comment (matteo-hertel) [inline packages/bpk-component-banner-alert/stories.js]: can we add the  `onDismiss` to log in the action logger console?

File: packages/bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 2]
```diff
@@ -101,9 +120,26 @@ class BpkBannerAlert extends Component {
     }));
   }
 
+  onDismiss() {
+    if (this.props.onDismiss) {
```
Comment (matteo-hertel) [inline packages/bpk-component-banner-alert/src/BpkBannerAlert.js]: you should probably check if it's a function with `typeof`

File: packages/bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 3]
```diff
@@ -127,7 +163,7 @@ class BpkBannerAlert extends Component {
     */
     // Disabling 'click-events-have-key-events and interactive-supports-focus' because header element is not focusable.
     // ToggleButton is focusable and works for this.
-    return (
+    return show ? (
```
Comment (matteo-hertel) [inline packages/bpk-component-banner-alert/src/BpkBannerAlert.js]: this nested ternary operators are really hard to read, can we refactor the logic a bit?
1. if `!show` return null straightaway 
2.  create a const for the expandible and dismissable first set them to null if no needed
3. return the rendered component adding inside the const created before

File: packages/bpk-component-banner-alert/src/BpkBannerAlert.js (hunk 2 above)
Comment (georgegillams) [reply packages/bpk-component-banner-alert/src/BpkBannerAlert.js]: Yeah I also had a chat to Matt about that, and he said that the type is enforced by the proptypes anyway. If it's not a function, react will blow up

**Comment by shaundon** [comment]
This is failing right on local and CI:

```ModuleNotFoundError: Module not found: Error: Can't resolve './AnimateAndFade' in '/Users/shaundonnelly/projects/backpack/packages/bpk-component-banner-alert/src'```

Just posting this here to remind myself to follow up on it next week.

File: packages/bpk-component-banner-alert/src/BpkBannerAlert.js [hunk 4]
```diff
@@ -62,7 +63,7 @@ const getIconForType = (type) => {
 
 const ToggleButton = (props) => {
   const classNames = [getClassName('bpk-banner-alert__expand-icon')];
-
+  console.log(props.expanded);
```
Comment (shaundon) [inline packages/bpk-component-banner-alert/src/BpkBannerAlert.js]: This should probably be removed before merging.

**Pull request merged**